Sources/KeyboardShortcuts/ViewModifiers.swift

```diff
@@ -0,0 +1,27 @@
+import SwiftUI
+
+@available(macOS 12, *)
+extension View {
+	public func onKeyboardShortcut(_ shortcut: KeyboardShortcuts.Name, perform: @escaping (KeyboardShortcuts.EventType) -> Void) -> some View {
+		self
+			.task {
+				for await _ in KeyboardShortcuts.on(.keyDown, for: shortcut) {
```

Comment:
I wonder if it makes sense to make it possible to not specify an event type for `.on(` too. That way we wouldn't need two `.task`'s here. The problem is that `.on(for: shortcut)` is ambiguous when you read it... "on" for what.

Reply:
That's something I wanted to raise in another issue or PR, a single stream that returns the event type could make sense. I had already written it actually:

```
	@available(macOS 10.15, *)
	public static func events(for name: Name) -> AsyncStream<KeyboardShortcuts.EventType> {
		AsyncStream { continuation in
			let id = UUID()

			streamKeyDownHandlers[name, default: [:]][id] = {
				continuation.yield(.keyDown)
			}

			streamKeyUpHandlers[name, default: [:]][id] = {
				continuation.yield(.keyUp)
			}

			registerShortcutIfNeeded(for: name)

			continuation.onTermination = { _ in
				streamKeyDownHandlers[name]?[id] = nil
				streamKeyUpHandlers[name]?[id] = nil

				unregisterShortcutIfNeeded(for: name)
			}
		}
	}
```

Reply:
Having two similar methods with completely different names kind bothers me though. "on" vs "events".

Can you think of any way we could have it consistent?

Reply:
I looked at other implementations that return an `AsyncStream` to get an idea of the grammar they use:
```
for try await line in url.lines
for try await data in publisher.values
```
The properties are usually named after the value that the stream collects, rather than when the collection happens (like `on(shortcut)` is).

My recommendation would be to use the `events` name, since it more closely follows the existing pattern with APIs using `AsyncStream`.

Thoughts?

Reply:
> (Maybe deprecate the existing one and provide two implementations, one that returns all the events and one that receives an EventType parameter and only sends those kinds of events)

Yeah. That's the best solution.

Reply:
@sindresorhus I made the changes.

I also ran into a crash that I think is unrelated (it happens on the new `events` function and on the existing `on` function). I don't know if it's something you've seen before:
```
-[NSTaggedPointerString count]: unrecognized selector sent to instance 0x8000000000000000
```

<details>
<summary>Full Stack Trace</summary>

```
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001b110ab08 __exceptionPreprocess + 240
	1   libobjc.A.dylib                     0x00000001b0e55e14 objc_exception_throw + 60
	2   CoreFoundation                      0x00000001b119ea68 -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x00000001b1069e14 ___forwarding___ + 1764
	4   CoreFoundation                      0x00000001b1069670 _CF_forwarding_prep_0 + 96
	5   libswiftCore.dylib                  0x00000001be1b35b0 $sSD8_VariantV8setValue_6forKeyyq_n_xtF + 120
	6   libswiftCore.dylib                  0x00000001be4386c8 $sSDyq_SgxcisTf4ngn_n + 428
	7   libswiftCore.dylib                  0x00000001be169f8c $sSDyq_Sgxcis + 32
	8   KeyboardShortcutsExample            0x0000000102184b00 $s17KeyboardShortcuts06CarbonaB0O8register_9onKeyDown0eF2UpyA2AO8ShortcutV_yAIcyAIctFZ + 844
	9   KeyboardShortcutsExample            0x000000010218b614 $s17KeyboardShortcutsAAO8register33_4B41ED0613F74375FB61C6AA00A4208CLLyyAB8ShortcutVFZ + 224
	10  KeyboardShortcutsExample            0x000000010218c708 $s17KeyboardShortcutsAAO24registerShortcutIfNeeded33_4B41ED0613F74375FB61C6AA00A4208CLL3foryAB4NameV_tFZ + 92
	11  KeyboardShortcutsExample            0x000000010218f144 $s17KeyboardShortcutsAAO6events3forScSyAB9EventTypeOGAB4NameV_tFZyScS12ContinuationVyAF_GXEfU_ + 1952
	12  libswift_Concurrency.dylib          0x0000000230a795b0 $sScS_15bufferingPolicy_ScSyxGxm_ScS12ContinuationV09BufferingB0Oyx__GyADyx_GXEtcfC + 252
	13  KeyboardShortcutsExample            0x000000010218e990 $s17KeyboardShortcutsAAO6events3forScSyAB9EventTypeOGAB4NameV_tFZ + 272
	14  KeyboardShortcutsExample            0x00000001021a8ddc $s7SwiftUI4ViewP17KeyboardShortcutsE02onD8Shortcut_4type7performQrA2DO4NameV_AH9EventTypeOyyctFyyYaYbcfU_ + 276
	15  KeyboardShortcutsExample            0x00000001021a9189 $s7SwiftUI4ViewP17KeyboardShortcutsE02onD8Shortcut_4type7performQrA2DO4NameV_AH9EventTypeOyyctFyyYaYbcfU_TATQ0_ + 1
	16  SwiftUI                             0x00000001d66ade95 $s7SwiftUI13_TaskModifierV05InnerD033_293A0AF83C78DECE53AFAAF3EDCBA9D4LLV4body7contentQrAA05_ViewD8_ContentVyAFG_tFyycfU_yyYaYbcfU_TQ0_ + 1
	17  SwiftUI                             0x00000001d66c8395 $s7SwiftUI13_TaskModifierV05InnerD033_293A0AF83C78DECE53AFAAF3EDCBA9D4LLV4body7contentQrAA05_ViewD8_ContentVyAFG_tFyycfU_yyYaYbcfU_TATQ0_ + 1
	18  SwiftUI                             0x00000001d66c2255 $sxIeghHr_xs5Error_pIegHrzo_s8SendableRzs5NeverORs_r0_lTRyt_Tg5TQ0_ + 1
	19  SwiftUI                             0x00000001d66c6221 $sxIeghHr_xs5Error_pIegHrzo_s8SendableRzs5NeverORs_r0_lTRyt_Tg5TATQ0_ + 1
	20  libswift_Concurrency.dylib          0x0000000230a95f09 _ZL23completeTaskWithClosurePN5swift12AsyncContextEPNS_10SwiftErrorE + 1
)
```

</details>

Reply:
> I also ran into a crash that I think is unrelated (it happens on the new events function and on the existing on function). I don't know if it's something you've seen before:

I have not seen that. How do I reproduce? Did it happen during a test run or usage?

Reply:
I'm seeing this quite consistently when running from Xcode. It doesn't happen when there are not shortcuts set yet.

Reply:
@sindresorhus It seems like it was a thread issue. I wrapped the event code in a dispatch to the main thread, and it no longer crashes.